src/plugins/intel_gpu/src/runtime/memory.cpp

```diff
@@ -38,6 +38,12 @@ memory::~memory() {
     }
```

Comment:
[random spot]
As I understand, this change assumes that constants will be always copied to device buffers (whether it's usm device/host, OpenCL buffer etc). But I think there are cases when memory do not need any modifications and then such constant will be passed directly as kernel argument and cause error, isn't it?

Reply:
I suppose `transfer_memory_to_device` function can ensure that no simple_attached_memory instances are left in the graph. Though I agree that this part requires careful check